--- TinyG-master/firmware/tinyg/plan_arc.c.original	2015-01-10 07:25:17.506916269 -0800
+++ TinyG-master/firmware/tinyg/plan_arc.c	2015-01-10 07:26:00.903191786 -0800
@@ -25,12 +25,11 @@
  * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF
  * OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
+#include <inttypes.h>
 #include <stdlib.h>
 #include <math.h>
 #include <stdio.h>				// precursor for xio.h
-#include <avr/pgmspace.h>		// precursor for xio.h
 
-#include "xio/xio.h"			// support trap and debug statements
 #include "tinyg.h"
 #include "config.h"
 #include "controller.h"			// only needed for line number
@@ -39,6 +38,7 @@
 #include "plan_arc.h"
 #include "planner.h"
 #include "kinematics.h"
+#include "stm32f4.h"
 
 /*
  * Local functions
